firefox-ios/Client/Frontend/Browser/Toolbars/AddressToolbarContainer.swift

```diff
@@ -13,7 +13,7 @@ protocol AddressToolbarContainerDelegate: AnyObject {
     func openSuggestions(searchTerm: String)
 }
 
-class AddressToolbarContainer: UIView,
+final class AddressToolbarContainer: UIView,
```

Comment:
Why is this class marked as `final`?

Reply:
The main reason is performance: When a class is marked as final, the compiler can perform certain optimizations, such as method dispatch optimization. Since Swift knows that the class cannot be subclassed, it can use more efficient dispatch mechanisms, like static dispatch, for methods rather than dynamic dispatch, that means that the complier wiil know which method to call at compile time. It is a good practice to declare every class that does not support inheritance, as `final`.